lib/vector-buffers/src/buffer_usage_data.rs

```diff
@@ -98,31 +110,26 @@ pub struct BufferUsageData {
     received_byte_size: AtomicU64,
     sent_event_count: AtomicU64,
     sent_byte_size: AtomicU64,
-    dropped_event_data: Option<BufferUsageDroppedEventData>,
+    dropped_event_count: AtomicU64,
+    dropped_event_byte_size: AtomicU64,
+    dropped_event_count_intentional: AtomicU64,
+    dropped_event_byte_size_intentional: AtomicU64,
     max_size_bytes: AtomicU64,
     max_size_events: AtomicUsize,
 }
 
-#[derive(Debug, Default)]
-struct BufferUsageDroppedEventData {
-    count: AtomicU64,
-    size: AtomicU64,
-}
-
 impl BufferUsageData {
-    pub fn new(mode: WhenFull, idx: usize) -> Self {
-        let dropped_event_data = match mode {
-            WhenFull::Block | WhenFull::Overflow => None,
-            WhenFull::DropNewest => Some(BufferUsageDroppedEventData::default()),
-        };
```

Comment:
We've killed the concept of buffers needing to specify up front if they can drop events because there's always a risk of unintentional event drops for any sufficiently complex buffer implementation.